tests/DCPS/Compiler/xcdr/xcdr.cpp

```diff
@@ -120,6 +161,12 @@ void expect_values_equal(const LC567Struct& a, const LC567Struct& b)
   EXPECT_EQ(a.ls[1], b.ls[1]);
 }
 
+template<typename TypeA, typename TypeB>
+void expect_values_equal_union(const TypeA& a, const TypeB& b)
+{
+  expect_values_equal_base_union(a, b);
+}
+
```

Comment:
Function template specializations for `expect_values_equal` are preferred. Per Fred's comments on my PR, the test is extended by defining template specializations for functions `set_values` and `expect_values_equal`. This function can be rewritten with something like:
```
template<>
void expect_values_equal(const MutableUnion& a, const MutableUnion& b)
```
for `MutableUnion` and another template specialization for `MutableXcdr12Union`.